schema.sql

```diff
@@ -48,14 +41,24 @@ CREATE TABLE specializations(
 	CONSTRAINT fk_vets FOREIGN KEY (vets_id) REFERENCES vets(id),
 	PRIMARY KEY (species_id, vets_id)
 );
-
 -- Animals-vets join table
 CREATE TABLE visits(
 	animals_id INT,
 	vets_id INT,
 	date_of_visit DATE,
 	CONSTRAINT fk_animals FOREIGN KEY (animals_id) REFERENCES animals(id),
 	CONSTRAINT fk_vets FOREIGN KEY (vets_id) REFERENCES vets(id),
-	CONSTRAINT visits_pk PRIMARY KEY
-	(animals_id, vets_id, date_of_visit)
+	CONSTRAINT visits_pk PRIMARY KEY (animals_id, vets_id, date_of_visit)
 );
+-- Added Email column to Owners table
+ALTER TABLE owners
+ADD COLUMN email VARCHAR(120);
+
+-- Added index to animals_id column on visits table
+CREATE INDEX idx_animal_id ON visits (animals_id ASC);
+
+-- Added index to vets_id column on visits table
+CREATE INDEX visits_covering_idx ON visits(vets_id) INCLUDE (visits_id, animals_id, vets_id, date_of_visit);
+
+--Create index to email column on owners table
+CREATE INDEX owner_email ON owners(email ASC);
```

Comment:
👏 **Optional Suggestion** 👏

Adding a case-insensitive index on the email column is a thoughtful consideration, especially when dealing with email addresses, which are not case-sensitive by convention. It allows for more flexible and user-friendly queries. To implement this, you can use the `lower()` function in your index creation:

```sql
CREATE INDEX ON owners ((lower(email)));
```

By doing this, you ensure that queries for emails are case-insensitive, making the database more user-friendly. However, please note that this suggestion is optional and might not be directly related to the performance issues you addressed in your project. It's more of an enhancement for user convenience.

Great job on being proactive in thinking about user experience and flexibility! Keep up the good work! 👍🌟

Reply:
Thank you for pointing this issue out. I'm quite new to databases and this is my first project so adding case insensitivity is not something I even thought about.  But now that you have mentioned It, I'll always have it in mind to implement it in future projects.

Reply:
You're very welcome! 😊

It's fantastic to hear that you found the suggestion helpful, especially in the context of your first project. Databases can indeed be a bit tricky at times, but you're on the right track by being open to new ideas and improvements.

Remember, learning and growing as a developer is all about exploring new concepts and continuously enhancing your skills. If you ever have more questions or need guidance in the future, don't hesitate to reach out. We're here to support your journey.

Best of luck with your future projects, and keep up the great work! 👍🚀

Reply:
Thank you very much for the encouragement, I appreciate it a lot!